synapse/rest/__init__.py

```diff
@@ -100,8 +100,7 @@ def register_servlets(client_resource: HttpServer, hs: "HomeServer") -> None:
         login.register_servlets(hs, client_resource)
         profile.register_servlets(hs, client_resource)
         presence.register_servlets(hs, client_resource)
-        if is_main_process:
-            directory.register_servlets(hs, client_resource)
+        directory.register_servlets(hs, client_resource)
```

Comment:
This is also loading `ClientAppserviceDirectoryListServer` and `ClientDirectoryListServer` on the workers.

At a skim, it looks fine, but it should then have a `CATEGORY` and it should be in the path map like the other endpoints?
If you don't have time to check that they're fine on workers, I'd add `if is_main_process` inside `directory.register_servlets` and leave those until later.

Reply:
> This is also loading ClientAppserviceDirectoryListServer and ClientDirectoryListServer on the workers.

I have it disabled.